src/Chapter3.hs

```diff
@@ -344,6 +346,15 @@ of a book, but you are not limited only by the book properties we described.
 Create your own book type of your dreams!
 -}
 
+data Genre = Fantasy | Novel | Thriller | SciFi | Other(String) deriving (Eq, Show)
```

Comment:
Wow, that's nice work on data types here! Very type-safe modelling :+1: 
I have only one minor stylistic suggestion. Usually, types are not take in `()` if they don't have type parameters 
```suggestion
data Genre = Fantasy | Novel | Thriller | SciFi | Other String
    deriving (Eq, Show)
```